src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -159,51 +184,94 @@ public static List<int> CommonBaudRates()
             };
         }
 
+        /// <summary>
+        /// Adds a new command handler.
+        /// A command handler can support extended commands.
+        /// </summary>
+        /// <typeparam name="T">An instance of <see cref="ExtendedCommandHandler"/>.</typeparam>
+        /// <param name="newCommandHandler">The new handler</param>
+        public void AddCommandHandler<T>(T newCommandHandler)
+            where T : ExtendedCommandHandler
+        {
+            if (newCommandHandler == null)
+            {
+                throw new ArgumentNullException(nameof(newCommandHandler));
+            }
+
+            _commandHandlersLock.EnterWriteLock();
```

Comment:
Nit: shouldn't the lock be entered after the if statement? I don't see where you are using _knownSupportedModes after the ifcheck so might as well move the entering of the lock one more line.

Reply:
`_extendedCommandHandlers` is also protected by the same lock. The two are updated synchronously.

Reply:
Shouldeach of them have their own lock?

Reply:
No, they're related and updated together. A new command handler may or may not bring additional modes, but since this is all done within the same method, there's no point in having two locks.